### PR TITLE
Context manager

### PR DESCRIPTION
**Pull request by tmichela**

### CODE REVIEW

File: karabo_bridge/client.py [hunk 1]
```diff
@@ -116,3 +116,15 @@ def _deserialize(self, msg):
             else:
                 raise RuntimeError('unknown message content:', md['content'])
         return dat
+
+    def __enter__(self):
+        return self
+    
+    def __exit__(self, exc_type, exc_value, traceback):
+        pass
```
Comment (takluyver) [inline karabo_bridge/client.py]: Should this close the socket or something? There's no point having a context manager if neither enter nor exit does anything.

**Comment by takluyver** [comment]
Just to make it clear, `for data in client` doesn't give you any way to catch an error and continue the loop - an exception would immediately break you out of the loop. Hopefully that's not a concern, but worth being aware of.

Also, I wonder if `for data in client` will make people expect that the loop exits when the data is 'finished' (e.g. run finished, or server shut down). Currently I think it will just hang indefinitely in that case. In a future revision of the protocol, it may make sense to add a way for the server to say that this is the end, and the client should not wait for more data.

File: karabo_bridge/client.py (hunk 1 above)
Comment (tmichela) [reply karabo_bridge/client.py]: True, maybe it is useless.
I also thought that the socket would be closed automatically when the context is excited, isn't it the case?

**Comment by tmichela** [comment]
This is a good point, there is a way to signal the end of data stream, and the server side is ready to sned it  (we would need to update the client). Unfortunately the DAQ is not using it at the moment... not sure if this is planned in near future

File: karabo_bridge/client.py (hunk 1 above)
Comment (takluyver) [reply karabo_bridge/client.py]: Not automatically from our point of view! All the `with` statement does is call these `__enter__` and `__exit__` methods. If we want it to have any effect, we have to write it in these methods.

It probably makes sense for it to close the socket on exit; I just checked, and `with ctx.socket() as sock:` will close on exiting the context as well. In most use cases I see, you'll exit the context when you're about to exit the program, so the socket should be cleaned up anyway. But it doesn't hurt to have an explicit way to ensure it's cleaned up.

**Comment by takluyver** [comment]
LGTM

**Pull request merged**